### ISSUE TITLE
texture() ignoring uvs for .obj file

### ISSUE THREAD

**Issue by kieranbrowne** [processing/p5.js#4866]
Hi there,

I'm trying to import a basic 3d .obj file and texture into p5js, but the texture appears to be ignoring the uv information in the obj file and misplacing the texture. I've written the equivalent program in processing 3 and it display correctly, so I can be pretty sure it's not an issue with the obj and texture files.

Am I doing something silly, or is this a genuine bug?

#### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Dom
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [x] WebGL
- [ ] Other (specify if possible)

#### Details about the bug:

- p5.js version: 1.1.9
- Web browser and version: Chrome: 86.0.4240.80 and Firefox: 81.0.2
- Operating System: MacOSX 10.15.6
- Steps to reproduce this: Load an obj file and texture into p5 with loadModel and loadImage.
The p5 code with all the necessary assets can be found here: <https://editor.p5js.org/kbrowne/sketches/o_PYbAgLl>

And here is the equivalent processing 3 code which textures correctly:
```
PShape mod;
PImage tex;

void setup() {
  size(400, 400, P3D);

  mod = loadShape("TreeFinger_L_03.obj");
  tex = loadImage("dif_arm.png");
  mod.setTexture(tex);
  noStroke();
}

void draw() {
  background(220);
  translate(width/2, height/2);
  scale(4);
  rotateX(frameCount * 0.01);
  rotateY(frameCount * 0.01);

  shape(mod, 0, 0);
}
```

**Comment by stalgiag** [comment]
Hi! I know that the [UVs are parsing correctly with some files](https://editor.p5js.org/stalgiag/sketches/ddWWhe1b1) but maybe there is a need to compare the UV parsing here to what is done in Processing to provide greater suport.

Anyone is welcome to take a look.

**Comment by kieranbrowne** [comment]
So I figured out that the y values for the uvs are inverted.

It's possible to reverse them like so:
```js
  for(let i=0; i< mod.uvs.length; i++) {
    mod.uvs[i][1] = 1.-mod.uvs[i][1];
  }
```

And this fixes the problem. See updated sketch: <https://editor.p5js.org/kbrowne/sketches/o_PYbAgLl>

This leaves open the question of why uvs appear to be parsed correctly for the file @stalgiag mentioned.
I rendered the capsule object and texture into processing and it displayed like so:

![processing](https://user-images.githubusercontent.com/5771172/97794775-1af39880-1c52-11eb-838a-5ad0edf71ef8.png)

But the equivalent code in p5 looks like this: 

![p5](https://user-images.githubusercontent.com/5771172/97794779-33fc4980-1c52-11eb-9f2f-882279b56991.png)

So it seems like p5 and processing are using the uv y values inversely.

**Comment by stalgiag** [comment]
Thank you @kieranbrowne! I think we should follow Processing on this. [It appears that](https://github.com/KhronosGroup/glTF-Sample-Viewer/issues/16#issuecomment-307239311) the lower-left pixel is `0,0` in most model format conventions and WebGL is the only one that needs to flip. This was likely not known when first implementing the loader.

**Comment by myselfhimself** [comment]
The added PR is ready for testing

**Comment by myselfhimself** [comment]
Here is a duplicated arm scene with a patched p5.js (uncompressed so 3.7MB)
https://editor.p5js.org/myselfhimself/sketches/yr8IQoK5I
This is indeed fixes the non-inverted V texture coordinate problem, and does not require you to for-loop and fix that in sketch.js

However, the capsule example with that patched p5.js file leads to a vertically flipped UV mapping visually:
https://editor.p5js.org/myselfhimself/sketches/V9ic4WvPg

Maybe we should all take care of how we export our .OBJ files... Blender3d's OBJ exporter for example allows to flip the forward and upward axes:
![image](https://user-images.githubusercontent.com/1265346/106207095-784fa500-61c1-11eb-9405-be34488c7963.png)
... and these settings do not work well for me.. my textures are always flipped in some way...

> Hi! I know that the [UVs are parsing correctly with some files](https://editor.p5js.org/stalgiag/sketches/ddWWhe1b1) but maybe there is a need to compare the UV parsing here to what is done in Processing to provide greater support.
> 
> Anyone is welcome to take a look.

Here is the related source in Processing (java), which states even in its preceding code commenting that the V values are inverted:
https://github.com/processing/processing/blob/94144be998354ecf7a06d5301b7b4fab72df724c/core/src/processing/core/PShapeOBJ.java#L215
```

          } else if (parts[0].equals("vt")) {
            // uv, inverting v to take into account Processing's inverted Y axis
            // with respect to OpenGL.
            PVector tempv = new PVector(Float.valueOf(parts[1]).floatValue(),
                                        1 - Float.valueOf(parts[2]).
                                        floatValue());
            texcoords.add(tempv);
            readvt = true;
```

So unless OpenGL and WebGL differ in behaviour and V x Y axes directions, I think that p5.js should follow Processing's behaviour and invert the V values...

**Comment by stalgiag** [comment]
closed with #5017
